### ISSUE TITLE
Option '-tex lualatex' makes an unexpected shift

### ISSUE THREAD

**Issue by lemniscati** [vectorgraphics/asymptote#123]
Asymptote 2.49 with LuaLaTeX 1.10 as engine on Cygwin makes an unexpected shift.
Is it an issue on Asymptote or on LuaLaTeX?

A test file `a.asy`:
```
real s=30;
draw(scale(s)*unitsquare,defaultpen+.5mm);
label("\TeX",scale(s)*(.5,.5));
```

Compiling as follows results in `default.pdf` and `lualatex.pdf`.

```
asy -f pdf -o default a
asy -f pdf -o lualatex -tex lualatex a
```

And, `default.pdf` has no problem, but `lualatex.pdf` has unexpected shift.

**Comment by johncbowman** [comment]
I tested this example with TeXLive 2019 and I can't see any difference between the two outputs.
I tried the current obsolete version Asymptote 2.49 and the latest stable version, 2.61, which I am just about to release...

**Comment by lemniscati** [comment]
Checking logs, and noticed the followings:

It seems to depend on whether `luaotfload` is successfully loaded through lualatex.

When `luaotfload-main.lua` is succesfully loaded on lualatex, it makes an unexpected shift.
But when `luaotfload-main.lua` is not found, it makes no unexpected shift.

Please, check `kpsewhich luaotfload-main.lua`.

Maybe, this issue is attributed to `luaotfload`...

**Comment by johncbowman** [comment]
Here is what I get (however, I have no idea what luaotfload-main is or how it might get loaded by your example; maybe there is a problem with your path?):

`/usr/local/texlive/2019/texmf-dist/tex/luatex/luaotfload/luaotfload-main.lua`

**Comment by johncbowman** [comment]
I do see the line:
luaotfload | main : initialization completed in 0.141 seconds
Can you post an example pdf that illustrates the shift you are seeing?

**Comment by lemniscati** [comment]
Thank you for testing.

Here is an example with unexpected shift
[lualatex.pdf](https://github.com/vectorgraphics/asymptote/files/3858024/lualatex.pdf)

and intermediate files
[lualatex_with_luaotfload-main.zip](https://github.com/vectorgraphics/asymptote/files/3858057/lualatex_with_luaotfload-main.zip)

**Comment by lemniscati** [comment]
And, here is an example without unexpected shift
[lualatex.pdf](https://github.com/vectorgraphics/asymptote/files/3858069/lualatex.pdf)

and intermediate files
[lualatex_without_luaotfload-main.zip](https://github.com/vectorgraphics/asymptote/files/3858071/lualatex_without_luaotfload-main.zip)

**Comment by lemniscati** [comment]
@johncbowman, please reopen the issue.

It can happen on Ubuntu 18.04 with TeX Live 2017, not only on Cygwin.

https://cygwin.com/ml/cygwin/2019-11/msg00137.html

And, on cygwin, I tried compiling from the source with ec3c7f09772ebc78880f7a08740e7009e57d20b8 (Asymptote version 2.62-3), and got same results (A logo TeX and an unit square are shitfted to the left with luaotfload-main.lua).

**Comment by johncbowman** [comment]
I could never see what was bothering you as the TeX label looks properly typeset and centered in the square in both examples that you posted. I now see that you are referring to a tiny horizontal offset of 0.4bp of the bounding box. It looks like a lualatex bug. I can change \hoffset in the generated texfile from -1in to -72.4bp as a temporary work around, until a proper fix is made in the lualatex sources.

**Comment by johncbowman** [comment]
A workaround was implemented in commit 41fbc67260971ebd5ab817ac902944d96462f543.
Please test that this works with and without luaotfload.

**Comment by lemniscati** [comment]
It works.
Thank you, @johncbowman!

I tested on cygwin in following cases. But I think the first case is important and sufficient, because both of  `luaotfload.sty` and  `luaotfload-main.lua` come with luatex/lualatex.

1. With `luaotfload.sty` and with `luaotfload-main.lua`: OK.
1. Without `luaotfload.sty` and with `luaotfload-main.lua`: OK.
1. With `luaotfload.sty` and without `luaotfload-main.lua`: shifted to the left.
1. Without `luaotfload.sty` and without `luaotfload-main.lua`: shifted to the left.
